DriverDataTimelineUI/src/main/java/com/drivequant/drivekit/timeline/ui/timeline/TimelineViewModel.kt

```diff
@@ -84,84 +74,62 @@ internal class TimelineViewModel(application: Application) : AndroidViewModel(ap
                 update()
             }
         }
-        DriveKitDriverData.getRawTimelines(
-            this.periods,
-            object : TimelineQueryListener {
-                override fun onResponse(
-                    timelineSyncStatus: TimelineSyncStatus,
-                    timelines: List<DKRawTimeline>
-                ) {
-                    if (timelineSyncStatus == TimelineSyncStatus.CACHE_DATA_ONLY) {
-                        timelines.forEach {
-                            when (it.period) {
-                                DKPeriod.WEEK -> weekTimeline = it
-                                DKPeriod.MONTH -> monthTimeline = it
-                                DKPeriod.YEAR -> {}
-                            }
-                        }
-                        update()
-                    }
+        DriveKitDriverData.getDriverTimelines(this.periods, SynchronizationType.CACHE, true) { timelineSyncStatus, timelines ->
+            if (timelineSyncStatus == TimelineSyncStatus.CACHE_DATA_ONLY) {
+                timelines.forEach {
+                    timelineByPeriod[it.period] = it
                 }
```

Comment:
On peut aussi améliorer en :
```suggestion
                timelineByPeriod = timelines.associateBy { it.period }
```
Et du coup transformer :
```
private var timelineByPeriod: MutableMap<DKPeriod, DKDriverTimeline> = mutableMapOf()
```
en :
```
private var timelineByPeriod: Map<DKPeriod, DKDriverTimeline> = mapOf()
```